framework/src/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -3106,7 +3106,7 @@ public boolean validateTypeOf(Tree tree) {
             }
             break;
         default:
-            type = atypeFactory.getAnnotatedType(tree);
+        	type = atypeFactory.getDefaultedAnnotatedType(tree);
```

Comment:
>To me using getDefaultedAnnotatedTypes here makes sense: take the type the programmer wrote for validation, without any refinements. This makes sense both for the LHS and for any other expression. 

I don't think this makes sense for any expression for example:

````
void foo(@D MyClass dParam, @C MyClass cParam, boolean flag) {
    @A MyClass d = dParam;
    @A MyClass c = cParam;
    @A MyClass cond = flag ? bRefined : cRefined;
}
````
Assuming `@B <: @A`, `@C <: @B`, `@D <: @B`, what type should getDefaultedAnnotatedType return given the conditional expression tree?